maven-core/src/main/java/org/apache/maven/execution/DefaultMavenExecutionResult.java

```diff
@@ -108,4 +110,16 @@ public void addBuildSummary( BuildSummary summary )
     {
         buildSummaries.put( summary.getProject(), summary );
     }
+
+    @Override
+    public boolean canResume()
+    {
+        return canResume;
+    }
+
+    @Override
+    public void setCanResume()
```

Comment:
Our style is pretty defensive: setter should have an argument. I can imagine we will boolean method references instead of if-true statements.

Reply:
Will add it, it sounds like a good idea. We can also simplify the if-true statement in `DefaultMaven#persistResumptionData` then